packages/web3-eth-contract/src/contract.ts

```diff
@@ -399,6 +412,35 @@ export class Contract<Abi extends ContractAbi>
 		return sendTransaction(this, tx, DEFAULT_RETURN_FORMAT);
 	}
 
+	private _contractMethodDeploySend<Options extends PayableCallOptions | NonPayableCallOptions>(
+		abi: AbiFunctionFragment,
+		params: unknown[],
+		options?: Options,
+		contractOptions?: ContractOptions,
+	) {
+		const tx = getSendTxParams({
+			abi,
+			params,
+			options,
+			contractOptions: contractOptions ?? this.options,
+		});
+
+		return sendTransaction(this, tx, DEFAULT_RETURN_FORMAT, {
+			transactionResolver: receipt => {
+				if (receipt.status === '0x0') {
+					throw new Web3ContractError('contract deployment error');
+				}
+
+				const newContract = this.clone();
```

Comment:
why we need to clone in case of successful deployment, as user will already have instance created, cant we set newly deployed address in that only.

Reply:
It's the behavior of the 1.x, after the deployment the new instance is returned.